clients/imodelhub/src/imodelhub/Briefcases.ts

```diff
@@ -279,8 +280,13 @@ export class BriefcaseHandler {
     ArgumentCheck.validGuid("iModelId", iModelId);
     ArgumentCheck.validBriefcaseId("briefcaseId", briefcaseId);
 
-    await this._imodelClient.locks.deleteAll(requestContext, iModelId, briefcaseId);
+    const existingLocksOwnedByBriefcase = await this._imodelClient.locks.get(requestContext, iModelId, new LockQuery().byBriefcaseId(briefcaseId).top(1));
```

Comment:
Probably we should have a way to delete briefcase without releasing locks separately. If user created some locks and lost Modify iModel permission, he won't be able to delete his briefcases.

Reply:
Added  a check for UserDoesNotHavePermission error to handle cases where user once had a modify permission but doesn't have that anymore